src/main/java/com/mineboundteam/minebound/magic/UtilitySpells/TelekineticUtilitySpell.java

```diff
@@ -74,24 +76,38 @@ public static void onPlayerTick(TickEvent.PlayerTickEvent event) {
         }
     }
 
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    public static void triggerSpell(LivingHurtEvent event) {
+        if (event.getEntityLiving() instanceof Player player && !player.getLevel().isClientSide()
+                && (event.getSource() == DamageSource.FALL || event.getSource() == DamageSource.FLY_INTO_WALL)) {
+            TelekineticUtilitySpell spell = (TelekineticUtilitySpell) getHighestEquippedSpellOfType(TelekineticUtilitySpell.class, player).getItem();
+            if (spell != null) {
+                event.setCanceled(true);
+                reduceMana(spell.manaCost, player);
+            }
+        }
+    }
```

Comment:
Working as intended; should be a LivingHurtEvent to prevent any additional resources being taken

Reply:
Ok, upon further testing I was able to get it to trigger on the `FLY_INTO_WALL` damage source, but not the `FALL` damage source. Were you able to get the event to fire with fall damage?

Reply:
Ok, upon further testing, I was able to trigger it with the `FLY_INTO_WALL` damage source, but not the `FALL` damage source. Were you able to fire this event with the `FALL` damage source with a telekinetic utility equipped?

Reply:
No, because of the canFly setting, but we might as well try and catch it in case some other mod invokes it manually, like how we damage the player using hunger. I'm not sure if canFly would catch that or not